### PR TITLE
HEXA-343 Read-only S3 buckets

### PR DESCRIPTION
**Pull request by ekatsah**
Note: test should pass if we merge #97 before + merge master. It helps us remove one migration and remove the need for a migration merge.

### CODE REVIEW

File: hexa/plugins/connector_s3/api.py [hunk 1]
```diff
@@ -152,23 +155,48 @@ def generate_sts_user_s3_credentials(
...
+    ro_bucket_names: typing.Optional[typing.Sequence[str]] = None,
+) -> typing.Dict:
+    statements = []
+
+    if ro_bucket_names:
+        statements.append(
+            {
+                "Sid": "S3ROActions",
+                "Effect": "Allow",
+                "Action": ["s3:ListBucket", "s3:GetObject*"],
```
Comment (pvanliefland) [inline hexa/plugins/connector_s3/api.py:169]: Are we sure that `s3:GetObject` (without the `*`) is not enough?

File: hexa/plugins/connector_s3/api.py [hunk 2]
```diff
@@ -64,9 +64,10 @@ def generate_sts_app_s3_credentials(
 def generate_sts_user_s3_credentials(
     *,
     principal_credentials: hexa.plugins.connector_s3.models.Credentials,
-    buckets: typing.Sequence[hexa.plugins.connector_s3.models.Bucket],
     role_identifier: str,
     session_identifier: str,
+    ro_buckets: typing.Sequence[hexa.plugins.connector_s3.models.Bucket],
```
Comment (pvanliefland) [inline hexa/plugins/connector_s3/api.py:69]: Let's use `read_only_buckets` and `read_write_buckets`

File: hexa/plugins/connector_s3/api.py [hunk 3]
```diff
@@ -152,23 +155,48 @@ def generate_sts_user_s3_credentials(
...
 
 
-def generate_s3_policy(bucket_names: typing.Sequence[str]) -> typing.Dict:
-    policy = {
-        "Version": "2012-10-17",
-        "Statement": [
+def generate_s3_policy(
+    rw_bucket_names: typing.Sequence[str] = None,
+    ro_bucket_names: typing.Optional[typing.Sequence[str]] = None,
+) -> typing.Dict:
```
Comment (pvanliefland) [inline hexa/plugins/connector_s3/api.py:161]: Same here, `read_only_buckets` and `read_write_buckets`

**Review by pvanliefland — CHANGES_REQUESTED** [review]
Almost ready imho, except for:

- A few (minor) change requests
- Adapt `test_credentials_s3fs` in `test_credentials.py` to check the mode
- We also miss a test that really verifies that we generate the proper policy depending on user bucket permissions

**Comment by ekatsah** [comment]
See #99

**Pull request closed without merging**